tests/integration/common/auth/test_auth.py

```diff
@@ -1,55 +1,103 @@
+        assert response.status_code == 302
+        assert response.location == url_for("auth.request_a_link_to_sign_in")
 
+    def test_redirect_on_used_magic_link(self, client, factories):
+        magic_link = factories.magic_link.create(
+            user__email="test@communities.gov.uk",
+            redirect_to_path="/my-redirect",
+            claimed_at_utc=datetime.datetime.now() - datetime.timedelta(hours=1),
+        )
 
-def test_sign_in_page_post_valid_email(client, mock_notification_service_calls):
-    response = client.post(
-        url_for("auth.sign_in"),
-        data={"email_address": "test@communities.gov.uk"},
-        follow_redirects=True,
-    )
-    assert response.status_code == 200
-    assert b"Check your email" in response.data
-    assert b"test@communities.gov.uk" in response.data
-    assert len(mock_notification_service_calls) == 1
-    with client.session_transaction() as sess:
-        assert sess["email_address"] == "test@communities.gov.uk"
+        response = client.get(url_for("auth.claim_magic_link", magic_link_code=magic_link.code))
+        assert response.status_code == 302
+        assert response.location == url_for("auth.request_a_link_to_sign_in")
 
+    def test_redirect_on_expired_magic_link(self, client, factories):
+        magic_link = factories.magic_link.create(
+            user__email="test@communities.gov.uk",
+            redirect_to_path="/my-redirect",
+            expires_at_utc=datetime.datetime.now() - datetime.timedelta(hours=1),
+        )
 
-def test_check_email_page_with_session(client):
-    with client.session_transaction() as sess:
-        sess["email_address"] = "test@communities.gov.uk"
+        response = client.get(url_for("auth.claim_magic_link", magic_link_code=magic_link.code))
+        assert response.status_code == 302
+        assert response.location == url_for("auth.request_a_link_to_sign_in")
 
-    response = client.get("/check-your-email")
-    assert response.status_code == 200
-    assert b"Check your email" in response.data
-    assert b"test@communities.gov.uk" in response.data
+    def test_post_claims_link_and_redirects(self, client, factories):
+        magic_link = factories.magic_link.create(
+            user__email="test@communities.gov.uk", redirect_to_path="/my-redirect", claimed_at_utc=None
+        )
 
+        response = client.post(
+            url_for("auth.claim_magic_link", magic_link_code=magic_link.code),
+            json={"submit": "yes"},
+            follow_redirects=False,
+        )
 
-def test_check_email_page_without_session(client):
-    response = client.get("/check-your-email", follow_redirects=True)
-    assert response.status_code == 200
-    assert b"Request a link to sign in" in response.data
-    assert url_for("auth.sign_in") in response.request.path
+        assert response.status_code == 302
+        assert response.location == "/my-redirect"
+        assert magic_link.claimed_at_utc is not None
```

Comment:
Very open to thoughts on whether we should be poking the magic link directly or mocking out the interface call. Or something else.

Reply:
I think this makes sense to me - if the code was redirecting but not appropriately setting this then the link could be claimed multiple times and it would impact the main purpose of the security, this feels more thorough than you might need to be in other instances but for this it's 👌? I'll have a bit more of a think